commit e8241313fc979aefafc8c2a0207570d6394f1580
Author: Martin Gräßlin <mgraesslin@kde.org>
Date:   Mon Jun 23 07:45:44 2014 +0200

    [libkwineffects] Require strict binding for intel driver
    
    According to driver devs it's a "buggy micro-optimisation", see
    https://bugs.freedesktop.org/show_bug.cgi?id=80349#c1
    
    BUG: 336589
    REVIEW: 118893

diff --git a/libkwineffects/kwinglplatform.cpp b/libkwineffects/kwinglplatform.cpp
index 4b2e6f3..9a391a0 100644
--- a/libkwineffects/kwinglplatform.cpp
+++ b/libkwineffects/kwinglplatform.cpp
@@ -811,7 +811,8 @@ void GLPlatform::detect(OpenGLPlatformInterface platformInterface)
             m_supportsGLSL = false;
 
         m_limitedGLSL = m_supportsGLSL && m_chipClass < I965;
-        m_looseBinding = true;
+        // see https://bugs.freedesktop.org/show_bug.cgi?id=80349#c1
+        m_looseBinding = false;
 
         if (m_chipClass < I915) {
             m_recommendedCompositor = XRenderCompositing;
